### PR TITLE
Explicitly check if the platform is supported

### LINKED ISSUE #7208: [BUG] Explicitly check if the platform is supported
**Describe the bug**
https://github.com/NVIDIA/spark-rapids/discussions/7205 was filed because a user tried to use the plugin with an unsupported GPU.  The error message was very confusing.  We should have some explicit check on startup before RMM runs that will verify that the GPU meets the minimum requirements and returns a clear error if it cannot.

### PR DESCRIPTION
**Pull request by cindyyuanjiang**
Resolves #7208 

We added a GPU device compute capability check in `RapidsExecutorPlugin`, and throws an exception with a clear message if the GPU is unsupported.

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala [hunk 1]
```diff
@@ -219,6 +219,10 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
       pluginContext: PluginContext,
       extraConf: java.util.Map[String, String]): Unit = {
     try {
+      if (Cuda.getComputeCapabilityMajor < 6) {
+        throw new RuntimeException(s"GPU compute capability ${Cuda.getComputeCapabilityMajor}" + 
+          " is unsupproted, requires 6.0+")
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala:224]:
```suggestion
          " is unsupported, requires 6.0+")
```

Comment (cindyyuanjiang) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala]: Updated. Thanks!

**Comment by cindyyuanjiang** [comment]
build

**Review by jlowe — APPROVED** [review]
Change looks good to me.  Is there a reason this is still in draft?  If so, it would be good to mention why.

**Comment by cindyyuanjiang** [comment]
> Change looks good to me. Is there a reason this is still in draft? If so, it would be good to mention why.

Thank you very much @jlowe! Not particularly, I will move this out of draft now.

**Review by revans2 — APPROVED** [review]

**Pull request merged**